File: projetIntegration-maha-backend/src/main/java/com/BoycottApp/BoycottApp/entities/Suggest.java
package com.BoycottApp.BoycottApp.entities;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@AllArgsConstructor
@NoArgsConstructor
@Data
public class Suggest {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @Enumerated(EnumType.STRING)
    private Type type;
    private String brandName;
    private String proofUrl;
    private String reasonWhy;
    private String alternativeOf;
    private String userName;
    private Boolean liked=false;


}

File: app/src/main/java/com/example/boycottini/Suggest.kt
package com.example.boycottini

data class Suggest(
    val id:Long,
    val type: String,         // "PRODUCT_TO_BOYCOTT" or "ALTERNATIVE"
    val brandName: String,
    val proofUrl: String,
    val reasonWhy: String?,
    val alternativeOf: String?,
    val userName: String,
    var liked: Boolean
)

